src/App.js

```diff
@@ -319,9 +302,28 @@ class App extends Component {
   );
 }
 
-const AppWithRedux = connect(state => ({
+const OldAppWithRedux = connect(state => ({
   walletReady: state.appState.walletReady,
-}))(App);
+}))(OldApp);
+
+function App() {
+  useAppState({
+    onChange(state) {
+      analytics.track('State change', {
+        category: 'app state',
+        label: state,
+      });
+    },
+    onActive() {
+      store.dispatch(walletConnectLoadState());
+      InteractionManager.runAfterInteractions(() => {
+        rainbowTokenList.update();
+      });
+    },
+  });
```

Comment:
Tbh, we might just want to be using React state for this type of thing e.g. `jotai` as part of our larger state work CC @sammdec

Reply:
Yeah agree with that!